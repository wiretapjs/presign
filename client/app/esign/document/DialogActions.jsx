import React from 'react'
import FlatButton from 'material-ui/FlatButton'
import PropTypes from 'prop-types'

const DialogActions = props => {
  return (
    <FlatButton label="CLOSE" type="button" style={{color: '#1F7FC7'}} onClick={props.onClick} />
  )
}

DialogActions.propTypes = {
  onClick: PropTypes.func,
}

export default DialogActions
