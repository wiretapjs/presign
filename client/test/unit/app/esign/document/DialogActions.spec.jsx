import React from 'react'
import { shallow } from 'enzyme'
import DialogActions from 'app/esign/document/DialogActions'

describe('DialogActions', () => {
  it('toggles openDialog after clicking on StyledLink', () => {
    const DialogActionsWrapper = shallow(<DialogActions />)
    expect(DialogActionsWrapper).toMatchSnapshot()
  })
})
