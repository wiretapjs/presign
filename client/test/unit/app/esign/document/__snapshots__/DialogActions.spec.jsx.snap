// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DialogActions toggles openDialog after clicking on StyledLink 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <FlatButton
    disabled={false}
    fullWidth={false}
    label="CLOSE"
    labelPosition="after"
    labelStyle={Object {}}
    onClick={undefined}
    onKeyboardFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onTouchStart={[Function]}
    primary={false}
    secondary={false}
    style={
      Object {
        "color": "#1F7FC7",
      }
    }
    type="button"
  />,
  "nodes": Array [
    <FlatButton
      disabled={false}
      fullWidth={false}
      label="CLOSE"
      labelPosition="after"
      labelStyle={Object {}}
      onClick={undefined}
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={[Function]}
      primary={false}
      secondary={false}
      style={
        Object {
          "color": "#1F7FC7",
        }
      }
      type="button"
    />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <DialogActions />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <FlatButton
          disabled={false}
          fullWidth={false}
          label="CLOSE"
          labelPosition="after"
          labelStyle={Object {}}
          onClick={undefined}
          onKeyboardFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          primary={false}
          secondary={false}
          style={
            Object {
              "color": "#1F7FC7",
            }
          }
          type="button"
        />,
        "_debugID": 2,
        "_renderedOutput": <FlatButton
          disabled={false}
          fullWidth={false}
          label="CLOSE"
          labelPosition="after"
          labelStyle={Object {}}
          onClick={undefined}
          onKeyboardFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          primary={false}
          secondary={false}
          style={
            Object {
              "color": "#1F7FC7",
            }
          }
          type="button"
        />,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <DialogActions />,
}
`;
